File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTrainingCPlus/Parsing/ProgramNode.cpp
#include "ProgramNode.h"
#include "Statements/IfThenElse/IfThenElseNode.h"

using namespace WorkflowsTraining::Parsing;
using namespace WorkflowsTraining::Parsing::Statements;

ProgramNode::ProgramNode(std::vector<std::unique_ptr<IStatementNode>>&& statements)
    : statements_(std::move(statements)) {
}

void ProgramNode::Execute() {
    for (const auto& statement : statements_) {
        statement->Execute();
    }
}

const std::vector<std::unique_ptr<IStatementNode>>& ProgramNode::GetStatements() const {
    return statements_;
}

File: WorkflowResults/WorkflowResults/Parsing/ProgramNode.cs
using WorkflowResults.Parsing.Interfaces;
using WorkflowResults.Parsing.Statements.Interfaces;

namespace WorkflowResults.Parsing;

public class ProgramNode(IList<IStatementNode> statements) : IAstNode
{
    private IList<IStatementNode> Statements { get; } = statements;

    public void Execute()
    {
        foreach (IStatementNode statement in Statements)
        {
            statement.Execute();
        }
    }
}